lib/services/access_list.go

```diff
@@ -206,6 +213,34 @@ func IsMember(identity tlsca.Identity, clock clockwork.Clock, accessList *access
 	return trace.NotFound("user %s is not a member of the access list", username)
 }
 
+// IsAccessListMember will return true if the user is a member for the current list.
+func IsAccessListMember(ctx context.Context, identity tlsca.Identity, clock clockwork.Clock, accessList *accesslist.AccessList, memberGetter AccessListMembersGetter) error {
+	username := identity.Username
+
+	member, err := memberGetter.GetAccessListMember(ctx, accessList.GetName(), username)
+	if trace.IsNotFound(err) {
+		// The member has not been found, so we know they're not a member of this list.
+		return trace.NotFound("user %s is not a member of the access list", username)
+	} else if err != nil {
+		// Some other error has occurred
+		return trace.Wrap(err)
+	}
+
+	expires := member.Spec.Expires
+	if expires.IsZero() || accessList.Spec.Audit.NextAuditDate.Before(expires) {
+		expires = accessList.Spec.Audit.NextAuditDate
+	}
+
+	if !clock.Now().Before(expires) {
+		return trace.NotFound("user %s's membership has expired in the access list", username)
```

Comment:
Should this be considered a not found error?
Once the membership expires should we use a `AccessDenied` instead?